packages/pluginutils/types/index.d.ts

```diff
@@ -0,0 +1,49 @@
+import { Node } from 'estree-walker';
+
+export interface AttachedScope {
+  parent?: AttachedScope;
+  isBlockScope: boolean;
+  declarations: { [key: string]: boolean };
+  addDeclaration(node: Node, isBlockDeclaration: boolean, isVar: boolean): void;
+  contains(name: string): boolean;
+}
+
+export interface DataToEsmOptions {
+  compact?: boolean;
+  indent?: string;
+  namedExports?: boolean;
+  objectShorthand?: boolean;
+  preferConst?: boolean;
+}
+
+export type AddExtension = (filename: string, ext?: string) => string;
+export const addExtension: AddExtension;
+
+export type AttachScopes = (ast: Node, propertyName?: string) => AttachedScope;
+export const attachScopes: AttachScopes;
```

Comment:
I’m not sure I understand the purpose of exporting an empty const here?

Reply:
This was actually generated by tsc haha

Reply:
Oh that makes sense. We can probably remove it in a future PR.